src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_pending_resumption_handler.cc

```diff
@@ -293,10 +316,114 @@ VehicleInfoPendingResumptionHandler::CreateSubscribeRequestToHMI(
 
   smart_objects::SmartObjectSPtr request =
       application_manager::MessageHelper::CreateModuleInfoSO(
-          hmi_apis::FunctionID::VehicleInfo_SubscribeVehicleData,
-          application_manager_);
+          VehicleInfo_SubscribeVehicleData, application_manager_);
   (*request)[strings::msg_params] = msg_params;
 
   return request;
 }
+
+smart_objects::SmartObject
+VehicleInfoPendingResumptionHandler::CreateFakeResponseFromHMI(
+    const std::map<std::string, smart_objects::SmartObject>& subscriptions,
+    const uint32_t fake_corrlation_id) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  namespace strings = application_manager::strings;
+
+  auto response =
+      application_manager::MessageHelper::CreateResponseMessageFromHmi(
+          VehicleInfo_SubscribeVehicleData,
+          fake_corrlation_id,
+          hmi_apis::Common_Result::SUCCESS);
+  auto& message = *response;
+  smart_objects::SmartObject msg_params(smart_objects::SmartType_Map);
+  for (const auto& subscription : subscriptions) {
+    msg_params[subscription.first] = subscription.second;
+    LOG4CXX_DEBUG(logger_,
+                  "fake response data : "
+                      << subscription.first << " result = "
+                      << subscription.second[strings::result_code].asInt());
+  }
+
+  message[strings::msg_params] = msg_params;
+  return *response;
+}
+
+bool VehicleInfoPendingResumptionHandler::PendingSubscriptionsResumption::
+    IsSuccessfullyDone() const {
+  return requested_vehicle_data_.size() == restored_vehicle_data_.size();
+}
+
+bool VehicleInfoPendingResumptionHandler::PendingSubscriptionsResumption::
+    DataWasRequested(const std::string& vd) const {
+  bool result =
+      (requested_vehicle_data_.end() != requested_vehicle_data_.find(vd));
+  return result;
+}
+
+std::set<std::string> VehicleInfoPendingResumptionHandler::
+    PendingSubscriptionsResumption::NotSubscribedData() const {
+  std::set<std::string> not_subscribed;
+  boost::set_difference(requested_vehicle_data_,
```

Comment:
@LuxoftAKutsan as I remember std contains the same algorithm. Why did you prefer boost? Just curious

Reply:
STD works with iterators and looks ugly. 
We are using boost in the project, o not see any reason not not use pretty boost function